php-checks/src/main/java/org/sonar/php/checks/IgnoredReturnValueCheck.java

```diff
@@ -0,0 +1,294 @@
+    "substr_compare",
+    "substr_count",
+    "substr_replace",
+    "substr",
+    "trim",
+    "ucfirst",
+    "ucwords",
+    "wordwrap",
+
+    // Array Functions
+    "array_change_key_case",
+    "array_chunk",
+    "array_column",
+    "array_combine",
+    "array_count_values",
+    "array_diff_assoc",
+    "array_diff_key",
+    "array_diff",
+    "array_fill_keys",
+    "array_fill",
+    "array_flip",
+    "array_intersect_assoc",
+    "array_intersect_key",
+    "array_intersect",
+    "array_key_exists",
+    "array_keys",
+    "array_merge_recursive",
+    "array_merge",
+    "array_pad",
+    "array_product",
+    "array_rand",
+    "array_replace_recursive",
+    "array_replace",
+    "array_reverse",
+    "array_search",
+    "array_slice",
+    "array_sum",
+    "array_unique",
+    "array_values",
+    "array",
+    "compact",
+    "count",
+    "current",
+    "in_array",
+    "key_exists",
+    "key",
+    "pos",
+    "range",
+    "sizeof",
+
+    // Variable handling Functions
+    "boolval",
+    "debug_zval_dump",
+    "doubleval",
+    "empty",
+    "floatval",
+    "get_defined_vars",
+    "get_resource_type",
+    "gettype",
+    "intval",
+    "is_array",
+    "is_bool",
+    "is_double",
+    "is_float",
+    "is_int",
+    "is_integer",
+    "is_iterable",
+    "is_long",
+    "is_null",
+    "is_numeric",
+    "is_object",
+    "is_real",
+    "is_resource",
+    "is_scalar",
+    "is_string",
+    "isset",
+    "serialize",
+    "strval",
+    "unserialize",
+
+    // Classes/Object Functions
+    "get_called_class",
+    "get_class_methods",
+    "get_class_vars",
+    "get_class",
+    "get_declared_classes",
+    "get_declared_interfaces",
+    "get_declared_traits",
+    "get_object_vars",
+    "get_parent_class",
+    "interface_exists",
+    "is_a",
+    "is_subclass_of",
+    "method_exists",
+    "property_exists",
+    "trait_exists",
+
+    // Ctype Functions
+    "ctype_alnum",
+    "ctype_alpha",
+    "ctype_cntrl",
+    "ctype_digit",
+    "ctype_graph",
+    "ctype_lower",
+    "ctype_print",
+    "ctype_punct",
+    "ctype_space",
+    "ctype_upper",
+    "ctype_xdigit",
+
+    // Filter Functions
+    "filter_has_var",
+    "filter_id",
+    "filter_input_array",
+    "filter_input",
+    "filter_list",
+    "filter_var_array",
+    "filter_var",
+
+    // Function handling Functions
+    "func_get_arg",
+    "func_get_args",
+    "func_num_args",
+    "function_exists"
+  ));
+
+  @Override
+  public void visitExpressionStatement(ExpressionStatementTree tree) {
+    checkExpression(tree.expression());
+    super.visitExpressionStatement(tree);
+  }
+
+  @Override
+  public void visitForStatement(ForStatementTree tree) {
+    tree.init().forEach(this::checkExpression);
+    tree.update().forEach(this::checkExpression);
+    super.visitForStatement(tree);
+  }
+
+  private void checkExpression(ExpressionTree expression) {
+    while (expression instanceof ParenthesisedExpressionTree) {
```

Comment:
I would use `expression.is(Tree.Kind.PARENTHESES_EXPRESSION)`. Or there is other trees which are ` parentheses?

Note that I introduced in this commit the method `skipParenthesis` (soon merged): https://github.com/SonarSource/sonar-php/pull/227/commits/005835315f8c1263ee992755db5868952f6477fd#diff-13bf78fd9141f2e4ac2a3d7e7ed9a996R123

Reply:
ok, I'm using: CheckUtils.skipParenthesis